### PR TITLE
Set up subsystem prongIO

### PR DESCRIPTION
**Pull request by countableclouds**
Prototype: Three Prong Gripper

This subsystem has one motor and one piston. The motor expands and contracts the three prongs, which can be used while contracted to collect a hatch. They can also be used, while expanded, to contract on a cargo. The piston is used to push out a collected cargo. Everything is set up, but ports, of course, will have to be modified accordingly for testing...

### CODE REVIEW

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
 
 public class RobotMap {
     public static class Port {
         public static class HumanInput {
             public static final int xboxController = 1;
 
         }
-        public static class CANMotor {}
+        public static class CANMotor {
+            //Define CANMotor.prongIOExpanderMotor with a correct port.
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: For Todos, add them on the line of the port, with `//TODO: Define with a correct port.`

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
+            public static final int prongIOExpanderMotor=-1;
+        }
         public static class PWM {}
         public static class CAN {}
-        public static class Pneumatics {}
+        public static class Pneumatics {
+            //Define Pneumatics.prongIOPusher with a correct port.
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: See my TODO-correction above.

File: src/main/java/org/usfirst/frc4904/robot/commands/PusherInside.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.usfirst.frc4904.robot.commands;
+
+
+public class PusherInside extends PusherSet {
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/commands/PusherInside.java]: Rename this to `PusherRetract`

File: src/main/java/org/usfirst/frc4904/robot/commands/PusherSet.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.usfirst.frc4904.robot.commands;
+import org.usfirst.frc4904.robot.RobotMap;
+import edu.wpi.first.wpilibj.command.Command;
+
+public class PusherSet extends Command {
+	protected final boolean fired;
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/commands/PusherSet.java:6]: rename `fired` to `fire`

File: src/main/java/org/usfirst/frc4904/robot/commands/PusherSet.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	}
+
+	@Override
+	protected void initialize() {
+		RobotMap.Component.prongIO.pusher.set(fired);
+	}
+
+	@Override
+	protected boolean isFinished() {
+		return false;
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/commands/PusherSet.java:21]: `return true;`

File: src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+
+import org.usfirst.frc4904.robot.commands.PusherInside;
+import org.usfirst.frc4904.standard.subsystems.motor.Motor;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+
+public class ProngIO {
+    //Will change speeds as needed.
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java:11]: Again, see the TODO-correction.

File: src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	public static final double EXPANSION_SPEED = 0.5;
+	public static final double CONTRACTION_SPEED = -0.5;
+	public final Motor expander;
+    public final Pusher pusher;
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java]: Fix your indentation!

File: src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+
+	public ProngIO(Motor expander, Pusher pusher) {
+		this.expander = expander;
+		this.pusher = pusher;
+	}
+
+	public static class Pusher extends Subsystem {
+		public static final DoubleSolenoid.Value FIRED = DoubleSolenoid.Value.kForward;
+		public static final DoubleSolenoid.Value INSIDE = DoubleSolenoid.Value.kReverse;
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java]: again, rename to RETRACT

File: src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+		protected final DoubleSolenoid pusher;
+
+		public Pusher(DoubleSolenoid pusher) {
+			super("ProngIO Pusher");
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java]: add an underscore or dash

File: src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+			this.pusher = pusher;
+		}
+
+		public void set(boolean fired) {
+			if (fired) this.pusher.set(FIRED);
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java]: Use regular if statements.

File: src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    public final Pusher pusher;
+
+
+	public ProngIO(Motor expander, Pusher pusher) {
+		this.expander = expander;
+		this.pusher = pusher;
+	}
+
+	public static class Pusher extends Subsystem {
+		public static final DoubleSolenoid.Value FIRED = DoubleSolenoid.Value.kForward;
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/subsystems/ProngIO.java:24]: add a TODO because we don't know the proper solenoid directions.

**Review by billwpierce — CHANGES_REQUESTED** [review]
A few small nitpicks, but nice job!

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 12]
```diff
@@ -1,23 +1,33 @@
...
 
 public class RobotMap {
     public static class Port {
         public static class HumanInput {
             public static final int xboxController = 1;
 
         }
-        public static class CANMotor {}
+        public static class CANMotor {
+            public static final int prongIOExpanderMotor=-1; //Define CANMotor.prongIOExpanderMotor with a correct port.
```
Comment (NikhilSuresh24) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: spacing please around equal sign

File: src/main/java/org/usfirst/frc4904/robot/commands/PusherSet.java (hunk 5 above)
Comment (countableclouds) [reply src/main/java/org/usfirst/frc4904/robot/commands/PusherSet.java:21]: About this:

What does this do?

**Pull request closed without merging**